src/Command/PqCommands/ContentHubPqEntityStructure.php

```diff
@@ -0,0 +1,230 @@
+
+  /**
+   * Paragraph field type.
+   */
+  public const PARAGRAPH_FIELD = 'paragraph';
+
+  /**
+   * Entity reference field type.
+   */
+  public const ENTITY_REF = 'entity_reference';
+
+  /**
+   * Entity reference revision field type.
+   */
+  public const ENTITY_REF_REV = 'entity_reference_revisions';
+
+  /**
+   * Formatted text field types.
+   */
+  public const FORMATTED_TEXT_FIELDS = [
+    'text_with_summary',
+    'text',
+    'text_long',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $defaultName = 'ach:pq:entity-structure';
+
+  /**
+   * The Drupal service factory service.
+   *
+   * @var \Acquia\Console\ContentHub\Command\Helpers\DrupalServiceFactory
+   */
+  protected $serviceFactory;
+
+  /**
+   * Constructs a new ContentHubPqEntityTypes object.
+   *
+   * @param \Acquia\Console\ContentHub\Command\Helpers\DrupalServiceFactory $drupalServiceFactory
+   *   The Drupal service factory service.
+   * @param string|null $name
+   *   The name of this command.
+   */
+  public function __construct(DrupalServiceFactory $drupalServiceFactory, string $name = NULL) {
+    parent::__construct($name);
+    $this->serviceFactory = $drupalServiceFactory;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function configure() {
+    parent::configure();
+    $this->setDescription('Analyses content type structure and sorts bundles based on their complexity.');
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * @throws \Exception
+   */
+  protected function runCommand(InputInterface $input, PqCommandResult $result): int {
+    $bundles = $this->getNodeBundles();
+    $fieldData = $this->getFieldTypes(array_keys($bundles));
+    $kriName = 'Risky Content Types';
+    $riskyBundles = $this->analyseFieldData($bundles, $fieldData);
+
+    $formatted = [];
+    foreach ($fieldData as $bundleId => $bundleData) {
+      $formatted[$bundleData['complex_fields']['count'] ?? 0] = sprintf('%s: %s ', $bundles[$bundleId], $bundleData['complex_fields']['count'] ?? 0);
+    }
+    ksort($formatted);
+
+    $formatted = array_merge(['Content Type: Number of Complex fields'], $formatted);
+
+    $kriMessage = PqCommandResultViolations::$riskyBundles;
+    $result->setIndicator(
+      $kriName,
+      implode("\n", $formatted),
+      $kriMessage
+    );
+
+    if (!empty($riskyBundles)) {
+      $result->setIndicator(
+        'Content types with paragraphs',
+        implode("\n", $riskyBundles),
+        PqCommandResultViolations::$paragraphBundles,
+        TRUE
+      );
+    }
+    return 0;
+
+  }
+
+  /**
+   * Returns node bundles.
+   *
+   * @return array
+   *   The list of bundle ids.
+   *
+   * @throws \Exception
+   */
+  public function getNodeBundles(): array {
+    $bundleIds = [];
+    /** @var \Drupal\Core\Entity\EntityTypeManager $etm */
+    $etm = $this->serviceFactory->getDrupalService('entity_type.manager');
+    $bundles = $etm->getStorage('node_type')->loadMultiple();
```

Comment:
If the entity type is dynamic we need to get the bundle name of it. This can be entity agnostic too.

Reply:
Bundle is available in the entity keys(in node's case: 'type') but the storage for bundle is different 'node_type'. In order to get it dynamically, need to create a separate logic, and since we were doing it just for nodes for now, I hardcoded it to node_type.